src/compiler/core.ts

```diff
@@ -1730,13 +1730,18 @@ namespace ts {
 
     /* @internal */
     export function startsWith(str: string, prefix: string): boolean {
-        return str.lastIndexOf(prefix, 0) === 0;
+        return str.indexOf(prefix) === 0;
+    }
+
+    /* @internal */
+    export function removePrefix(str: string, prefix: string | undefined): string {

```

Comment:
`startsWith` works with an undefined prefix. I'll change the annotation there as well.